src/main/java/hudson/plugins/gradle/injection/GradleBuildScanInjection.java

```diff
@@ -33,18 +36,34 @@ public String getActivationEnvironmentVariableName() {
     public void inject(Node node, EnvVars envGlobal, EnvVars envComputer) {
         try {
             String initScriptDirectory = getInitScriptDirectory(envGlobal, envComputer);
-            if (isOn(envGlobal)) {
-                copyInitScript(envGlobal, envComputer, node.getChannel(), initScriptDirectory);
-            } else {
-                removeInitScript(node.getChannel(), initScriptDirectory);
+
+            removeInitScript(node.getChannel(), initScriptDirectory);
```

Comment:
Wouldn't it lead to the injection every time the node is restarted? If in the future we will be downloading files, it will be more fragile.

Reply:
As discussed we need to handle versioning and up to date checks in a separate PR for both.